controller/dataplane/controller.go

```diff
@@ -181,6 +181,11 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		labelSelectorFromDataPlaneStatusSelectorDeploymentOpt(dataplane),
 	}
 
+	// if the dataplane is configured with Konnect, the status/ready endpoint should be set as the readiness probe.
+	if _, konnectApplied := k8sutils.GetCondition(consts.KonnectExtensionAppliedType, dataplane); konnectApplied {
+		deploymentOpts = append(deploymentOpts, statusReadyEndpointDeploymentOpt(dataplane))
+	}
```

Comment:
I believe this will also have to be added to the BG controller deployment options at https://github.com/Kong/gateway-operator/blob/292b536f7ff2f43eac7ce99b8668fee6180f4efa/controller/dataplane/bluegreen_controller.go#L515-L517

Reply:
Good catch. See 1755237007f990775033287098e80eaf7656a07f